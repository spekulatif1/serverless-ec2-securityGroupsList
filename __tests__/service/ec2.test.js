'use strict'

const {
	EC2Client,
	DescribeSecurityGroupsCommand
} = require("@aws-sdk/client-ec2");
const { mockClient } = require("aws-sdk-client-mock");
const ec2Mock = mockClient(EC2Client);
const { ec2 } = require('../../service');
const { awsSecurityGroupsListResponse } = require('../../data/sample');

describe('ec2.service', () => {
	beforeEach(() => {
		ec2Mock.reset();
	});

	const TEST_TOKEN = 'TEST';
	test('ec2.service.securityGroups.success', async () => {
		ec2Mock
			.on(DescribeSecurityGroupsCommand, { NextToken: undefined })
			.resolves(awsSecurityGroupsListResponse)

		const successResponse = await ec2.listSecurityGroups();

		expect(successResponse).toMatchSnapshot();
		expect(ec2Mock.calls()).toHaveLength(1);
	});

	test('ec2.service.securityGroups.success.with.NextToken', async () => {
		ec2Mock
			.on(DescribeSecurityGroupsCommand, { NextToken: undefined })
			.resolves({...awsSecurityGroupsListResponse, NextToken: TEST_TOKEN })
			.on(DescribeSecurityGroupsCommand, { NextToken: TEST_TOKEN })
			.resolves({...awsSecurityGroupsListResponse, NextToken: undefined });

		const successResponse = await ec2.listSecurityGroups();

		expect(successResponse).toMatchSnapshot();
		expect(ec2Mock.calls()).toHaveLength(2);
	});
});
