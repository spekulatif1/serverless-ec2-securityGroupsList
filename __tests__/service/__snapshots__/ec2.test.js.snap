// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ec2.service ec2.service.securityGroups.success 1`] = `
Object {
  "body": "{\\"data\\":[{\\"Description\\":\\"TrendMicro-Assignment-Test\\",\\"GroupName\\":\\"TrendMicro-Assignment-Test\\",\\"IpPermissions\\":[{\\"FromPort\\":80,\\"IpProtocol\\":\\"tcp\\",\\"IpRanges\\":[{\\"CidrIp\\":\\"0.0.0.0/0\\"}],\\"Ipv6Ranges\\":[{\\"CidrIpv6\\":\\"::/0\\"}],\\"PrefixListIds\\":[],\\"ToPort\\":80,\\"UserIdGroupPairs\\":[]}],\\"OwnerId\\":\\"TrendMicro\\",\\"GroupId\\":\\"sg-trendMicro\\",\\"IpPermissionsEgress\\":[{\\"IpProtocol\\":\\"-1\\",\\"IpRanges\\":[{\\"CidrIp\\":\\"0.0.0.0/0\\"}],\\"Ipv6Ranges\\":[],\\"PrefixListIds\\":[],\\"UserIdGroupPairs\\":[]}],\\"VpcId\\":\\"vpc-trendMicro\\"}]}",
  "statusCode": 200,
}
`;

exports[`ec2.service ec2.service.securityGroups.success.with.NextToken 1`] = `
Object {
  "body": "{\\"data\\":[{\\"Description\\":\\"TrendMicro-Assignment-Test\\",\\"GroupName\\":\\"TrendMicro-Assignment-Test\\",\\"IpPermissions\\":[{\\"FromPort\\":80,\\"IpProtocol\\":\\"tcp\\",\\"IpRanges\\":[{\\"CidrIp\\":\\"0.0.0.0/0\\"}],\\"Ipv6Ranges\\":[{\\"CidrIpv6\\":\\"::/0\\"}],\\"PrefixListIds\\":[],\\"ToPort\\":80,\\"UserIdGroupPairs\\":[]}],\\"OwnerId\\":\\"TrendMicro\\",\\"GroupId\\":\\"sg-trendMicro\\",\\"IpPermissionsEgress\\":[{\\"IpProtocol\\":\\"-1\\",\\"IpRanges\\":[{\\"CidrIp\\":\\"0.0.0.0/0\\"}],\\"Ipv6Ranges\\":[],\\"PrefixListIds\\":[],\\"UserIdGroupPairs\\":[]}],\\"VpcId\\":\\"vpc-trendMicro\\"},{\\"Description\\":\\"TrendMicro-Assignment-Test\\",\\"GroupName\\":\\"TrendMicro-Assignment-Test\\",\\"IpPermissions\\":[{\\"FromPort\\":80,\\"IpProtocol\\":\\"tcp\\",\\"IpRanges\\":[{\\"CidrIp\\":\\"0.0.0.0/0\\"}],\\"Ipv6Ranges\\":[{\\"CidrIpv6\\":\\"::/0\\"}],\\"PrefixListIds\\":[],\\"ToPort\\":80,\\"UserIdGroupPairs\\":[]}],\\"OwnerId\\":\\"TrendMicro\\",\\"GroupId\\":\\"sg-trendMicro\\",\\"IpPermissionsEgress\\":[{\\"IpProtocol\\":\\"-1\\",\\"IpRanges\\":[{\\"CidrIp\\":\\"0.0.0.0/0\\"}],\\"Ipv6Ranges\\":[],\\"PrefixListIds\\":[],\\"UserIdGroupPairs\\":[]}],\\"VpcId\\":\\"vpc-trendMicro\\"}]}",
  "statusCode": 200,
}
`;
